### PR TITLE
Feat: specify initial concentrations individually for the ODE state vector

### PR DESCRIPTION
**Pull request by carrascomj**
Supersedes  #399 

I have been using this feature locally for a while and it is very useful when the concentrations in the model may span over various orders of magnitudes.

Instead of being set on the `config.toml` file, initial concentrations are set as any other balanced mic measurement, specifying a field `likelihood=False`.  The semantic is _"use this value but do not use it to compute the likelihood"_. That may be a bit confusing, let me know what think before I add documentation.

Checklist:

- [x] Updated any relevant documentation
- [ ] Add an adr doc if appropriate
- [ ] Include links to any relevant issues in the description
- [x] Unit tests passing
- [x] Integration tests passing

### CODE REVIEW

**Comment by teddygroves** [comment]
I think this is a really important feature but I'm not sure about this way of implementing it.

If I follow correctly it depends on Maud setting initial states based on the measured concentration where available: this might not be obvious to users and also might change in future.

I think it might be better to replace the current system with having custom initial states in their own field under "experiment". Something like:

```toml
[[experiment]]
id = "dataset1"
is_train = true
is_test = false
initial_state = [
 {metabolite = "5mthf", compartment = "c", initial_state = 5.79e-06},
]

```
This should result in an overall simpler implementation than what we currently have, and is also easier to explain. The downside is it makes the input a bit more verbose, but I think it's worth it. Others who have done more Maud modelling than me probably know better though - let me know what you think!

**Comment by carrascomj** [comment]
Yes, that makes more sense. I implemented that but with "value" instead of "initial_state":

```toml
initial_state = [
 {metabolite = "5mthf", compartment = "c", value = 5.79e-06},
]
```

since it makes it easier to move from a measurement to a initial_state and otherwise it sounds redundant (`initial_state[0].value` instead of `initial_state[0].initial_state`).

File: docs/usage/inputting.rst [hunk 1]
```diff
@@ -353,6 +354,19 @@ The measurement table has these fields:
 measurements or the scale parameter of a lognormal distribution for
 concentration measurements.
 
+The initial_state table has these fields:
+
+- :code:`metabolite` A string identifying the metabolite.
+- :code:`compartment` A string identifying the compartment.
+- :code:`value` The measured value, as a float.
```
Comment (teddygroves) [inline docs/usage/inputting.rst:361]: measured value -> initial state

File: docs/usage/inputting.rst [hunk 2]
```diff
@@ -353,6 +354,19 @@ The measurement table has these fields:
 measurements or the scale parameter of a lognormal distribution for
 concentration measurements.
 
+The initial_state table has these fields:
```
Comment (teddygroves) [inline docs/usage/inputting.rst:357]: initial_state -> :code:`initial_state`

File: docs/usage/inputting.rst [hunk 3]
```diff
@@ -353,6 +354,19 @@ The measurement table has these fields:
...
+
+- :code:`metabolite` A string identifying the metabolite.
+- :code:`compartment` A string identifying the compartment.
+- :code:`value` The measured value, as a float.
+
+A default initial concentration for all metabolites and experiments can
+be specified in the general configuration. In addition, this initial_state
```
Comment (teddygroves) [inline docs/usage/inputting.rst:364]: initial_state -> :code:`initial_state`

**Review by teddygroves — CHANGES_REQUESTED** [review]
Looks great! 

I think for simplicity we should eventually retire the current method based on the measured value, but that can wait.

Just found a couple of typos in the new documentation.

File: docs/usage/inputting.rst (hunk 3 above)
Comment (carrascomj) [reply docs/usage/inputting.rst]: Done!

**Review by teddygroves — APPROVED** [review]

**Pull request merged**